### DISCUSSION TITLE
dag.is_paused web ui/database out of sync

### DISCUSSION THREAD

**Discussion by gmi-joelbondurant** [apache/airflow#29249] [General]
### Apache Airflow version

2.3.3

### What happened

I've repeatedly encountered cases where the web ui and database are not in sync regarding the dag.is_paused state. Flipping the dag is_paused button usually works 99% of the time, but is definitely not 100% accurate. I was just notified about a production dag which wasn't running and the audit logs show the is_paused state to be the opposite of the web ui. I've seen this bug happening for years.

![image](https://user-images.githubusercontent.com/103617862/215557237-c15d3ae6-345e-40cc-9b6d-47197932a7e0.png)

### What you think should happen instead

The dag.is_paused state as rendered by the Airflow web ui is not a reliable representation
of the actual dag.is_paused state, but it should be, please consider synchronous blocking transactions.

### How to reproduce

Flip thousands of dags on or off rapidly with multiple Airflow windows open, error rates between 0.1% and 1% will occur. To enhance the failure rate, filter dags by paused state "Active" and refresh the page periodically while doing this.

### Operating System

Linux

### Versions of Apache Airflow Providers

Airflow 2.3.3
Google Cloud Composer 2.0.30

### Deployment

Composer

### Deployment details

 Resources
Workloads configuration
Scheduler
    2 vCPUs, 8 GB memory, 4 GB storage 
Number of schedulers
    1 
Web server
    1 vCPU, 3 GB memory, 4 GB storage 
Worker
    10 vCPUs, 64 GB memory, 10 GB storage 
Number of workers
    Autoscaling between 1 and 4 workers 
Core infrastructure
Environment size
    Small 

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Not nearly good enough for reproducible case. Converting it into discussion.

I would find it extremely strange to have Airflow showing different state of DAG than the one that is in the DB, because the state displayed in UI is reflecting the state of the DB **at the moment the page in your browser has been refreshed** (note this specific condition).

Refreshing a UI is going to be always asychronous to changes of state, this is not at all strange. We do not refresh the state as it happens on the database in this case (it would require - contrary to the ask - asynchronous communication and websocket interface between the server and UI).

Unless I do not understand your problem description - In order to get the right state of the DAG you always have to refresh the screen you are looking at. For example when someone changes the state to "paused" in their UI, you need to refresh your UI to see it. This how Airlfow currently works. 

If you would like all the UIs to get refreshed immediately after any of the DAG states change by another session then this is a feature request not a DAG (and one requiring websockets and generally asynchronous communication between the client and webserver that we currently do not have). We debated whether we want it and rejected it, and instead we opted for "auto-refresh" feature in DAGs page that likely you have disabled that implements the refresh periodically. If you have it disabled that might be the reason you see the problem.

The question is - is the paused state going back to the right state after you refresh the DAGs list? In other words is this more than temporary disconnect between what you see on previously rendered page and the database? If the state is fixed after refreshing - all is fine. If it stays different than you expect it  and your state in the DB shows "paused" and UI state is "unpaused" even when you  refresh the  UI few times, then we would have a bug. In this case - can you please capture the state in the database as different than the one you see, and add some logs from the webserver showing the refresh of your page and fetching the wrong value with debug logs enabled? (you can enable debug logging via configuration).,

There is potentially another reason for that - in case you have delayed replication of the Database of yours and your webserver is accessing the delayed replica, you migt see similar behaviour. But Airlfow does not support such setup and it's more of a deployment issue than Airlfow issue.,

**Reply by potiuk** [reply]
One more comment. Your issue report mislead me @gmi-joelbondurant 

When reporting the issue you wrongly indicated Airflow 2.5.1 in your issue and later you mentioned 2.3.3 (you should choose "other").
 If this is 2.3.3 - then DAG auto refresh was I think not implemented back then. You should upgrade. Then your DAGs will be auto-refreshing periodically when you will have the page open and your problem will likely be gone.
